### PR TITLE
Extract spl-math-utils from spl-math

### PR DESCRIPTION
**Pull request by kevinheavey**
Crates that are using `spl-math` as a library don't need the parts that use `solana_program` .

This PR pulls the math parts into a separate `spl-math-utils` crate and updates `spl-token-swap` to use that instead of `spl-math`. That's the only SPL crate that uses `spl-math`.

This PR also removes `spl-math` from the `spl-stake-pool` deps but `spl-stake-pool` wasn't using `spl-math` in the first place

### CODE REVIEW

**Comment by kevinheavey** [comment]
Clippy errors are unrelated to this PR:

```
error: doc list item without indentation
  --> token/program-2022/src/extension/confidential_mint_burn/instruction.rs:80:9
   |
80 |     ///   3.. `[signer]` Required M signer accounts for the SPL Token Multisig
```

File: libraries/math-utils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "spl-math-utils"
```
Comment (joncinque) [inline libraries/math-utils/Cargo.toml:2]: Can we have this crate be `spl-math` and change the one with the program to `spl-math-example`?

**Review by joncinque — COMMENTED** [review]
Great change, thanks! Since it's more of a mistake that spl-math included a whole a program, can we rename the program bit instead?

File: libraries/math-utils/Cargo.toml (hunk 1 above)
Comment (kevinheavey) [reply libraries/math-utils/Cargo.toml]: Done. Set the spl-math version back to 0.3 and set spl-math-example version to 0.1

**Review by joncinque — APPROVED** [review]
Thanks!

**Pull request merged**